pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+        self.instances = {}
+
+        # Open Pull Contexts. The key for each context is an enumeration
+        # context id.  The data is the total list of instances/names to
+        # be returned and the current position in the list. Any context in
+        # this list is still open.
+        self.enumeration_contexts = {}
+
+        self.verbose = verbose
+
+        self.logfile = None
+
+        self._imethodcall = Mock(side_effect=self._mock_imethodcall)
+        self._methodcall = Mock(side_effect=self._mock_methodcall)
+
+    ##########################################################
+    #
+    #   Functions Mocked
+    #
+    ##########################################################
+
+    def _mock_imethodcall(self, methodname, namespace, response_params_rqd=None,
+                          **params):
+        """
+        Mock the WBEMConnection _imethod call.
+
+        This mock calls functions within this class that fake the processing
+        in a WBEM server (at the CIM Object level) for the varisous CIM/XML
+        methods and return.
+
+        Each function is named with the lower case method namd prepended with
+        'fake_'.
+        """
+        if self.verbose:
+            print('mock_imethodcall method=%s, namespace=%s, '
+                  'response_params_rqd=%s\nparams=%s'
+                  % (methodname, namespace, response_params_rqd, params))
+
+        method_name = '_fake_' + methodname.lower()
+
+        method_name = getattr(self, method_name)
+        result = method_name(namespace, **params)
+
+        if self.verbose:
+            print('mock result %s' % result)
+
+        return result
+
+    def _mock_methodcall(self, methodname, localobject, Params=None, **params):
+        # pylint: disable=invalid-name
+        """
+        Mock the execution of WBEMConnection _methodcall which is the common
+        calling interface for invokemethod
+        """
+        if self.verbose:
+            print('mock_imethodcall method=%s, namespace=%s, '
+                  'response_params_rqd=%s\nparams=%s'
+                  % (methodname, localobject, Params, params))
+
+        result = self._fake_invokemethod(methodname,
+                                         localobject,
+                                         Params=Params,
+                                         **params)
+
+        if self.verbose:
+            print('mock result %s' % result)
+
+    ################################################################
+    #
+    #   Methods to insert data into fake repository
+    #
+    ################################################################
+
+    @staticmethod
+    def display_objects(obj_type, objects_dict):
+        """
+        Display a set of objects of obj_type from the dictionary defined
+        by the parameter objects_dict. obj_type is a string that defines the
+        type of object (instance, class, qualifier declaration).
```

Comment:
If this method is supposed to be available for the user, document its parameters with a "Parameters:" section and list of "name (type): description".

Reply:
This one moved to internal.  We will externalize the general display_repository with more parameters and hide this method.

Reply:
DISCUSS: Now shown in list of discussion points, below.